content/courses/ada_2022_whats_new/chapters/delta_aggregates.rst

```diff
@@ -0,0 +1,99 @@
+.. include:: ../../global.txt
+
+Delta Aggregates
+================
+
+.. note::
+
+    Delta aggregates are supported by
+
+    * GNAT Community Edition 2019
+    * GCC 9
+
+Sometimes you need to create a copy of an object with a few
+modifications. Before Ada 2022 it would involve a dummy object
+declaration or an aggregate with associations for each property.
+The dummy object approach doesn't work in contract aspects.
+A limited component won't work with dummy object approach neither.
+On the other side re-listing properties in an aggregate could be
+too hard. So, in Ada 2022, you can use a `delta aggregates` instead.
+
+Delta aggregate for records
+---------------------------
+
+The delta aggregate for a record type looks like this:
+
+.. code:: ada no_button manual_chop project=Courses.Ada_2022_Whats_New.Delta_Aggregates.Record
+
+   !main.adb
+   type Vector is record
```

Comment:
In this specific case, you could create an actual package and use the `compile_button`.